pool.go

```diff
@@ -451,6 +540,48 @@ func (p *pool) fetchAndUpdate(ctx context.Context, node string, c cid.Cid, attem
 	return
 }
 
+func (p *pool) fetchResourceAndUpdate(ctx context.Context, node string, path string, attempt int, cb DataCallback, transientErrs map[string]error) (err error) {
+	err = p.fetchResource(ctx, node, path, "application/vnd.ipld.car", attempt, cb)
+	if err != nil {
+		goLogger.Debugw("fetch attempt failed", "from", node, "attempt", attempt, "of", path, "error", err)
+	}
+
+	if err == nil {
+		p.changeWeight(node, false)
+		// Saturn fetch worked, we return the block.
+		return
+	}
+
+	// If this is a NOT found or Timeout error, park the downvoting for now and see if other members are able to give us this content.
+	if errors.Is(err, ErrContentProviderNotFound) || errors.Is(err, ErrSaturnTimeout) || errors.Is(err, ErrPartialResponse{}) {
+		transientErrs[node] = err
+		return
+	}
+
+	// Saturn fetch failed, we downvote the failing member.
+	p.changeWeight(node, true)
+	return
+}
+
+type weightUpdateReq struct {
+	node    string
+	failure bool
+}
+
+func (p *pool) updateWeightBatched(reqs []weightUpdateReq) {
```

Comment:
Please can you add a test for `updateWeightBatched`

Reply:
you added (and maybe later removed) this
https://github.com/filecoin-saturn/caboose/blob/67d62dd0ca18af3137522a6ae103b538f2923cb3/pool.go#L329

Reply:
Okay I'll add the test back.